src/SonarScanner.MSBuild.PreProcessor/SonarWebService.cs

```diff
@@ -273,9 +253,8 @@ public Task<AnalysisCacheMsg> DownloadCache(string projectKey, string branch)
             return downloader.DownloadStream(uri).ContinueWith(x => AnalysisCacheMsg.Parser.ParseFrom(x.Result));
         }
 
-        #endregion ISonarQubeServer interface
-
-        #region Private methods
+        private async Task<bool> IsSonarCloud() =>
+            SonarProduct.IsSonarCloud(serverUri.Host, await GetServerVersion());
```

Comment:
Line 271:
Let's rename `op` to `action`.
I also propose to rename `uri` to `logUri`, because it's used only for logging. And it's confusing on the invocation side to understand why is the `uri` often used twice
```
private async Task<T> DoLogExceptions<T>(Func<Task<T>> action, Uri logUri)
```